--- origsrc/SDL_image-1.2.10/configure.in	2009-11-12 00:41:21.000000000 -0600
+++ src/SDL_image-1.2.10/configure.in	2009-11-17 01:41:15.633382900 -0600
@@ -61,7 +61,7 @@ case "$host" in
     *-*-beos*)
         ac_default_prefix=/boot/develop/tools/gnupro
         ;;
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-mingw32*)
         if test "$build" != "$host"; then # cross-compiling
             # Default cross-compile location
             ac_default_prefix=/usr/local/cross-tools/i386-mingw32
@@ -166,7 +167,10 @@ if test x$enable_jpg = xyes || test x$en
             *-*-darwin*)
                 jpg_lib=[`find_lib libjpeg.dylib`]
                 ;;
-            *-*-cygwin* | *-*-mingw32*)
+            *-*-cygwin*)
+                jpg_lib="cygjpeg-8.dll"
+                ;;
+            *-*-mingw32*)
                 jpg_lib=[`find_lib "jpeg.dll"`]
                 ;;
             *)
@@ -201,7 +205,10 @@ if test "x$enable_png" = xyes; then
             *-*-darwin*)
                 png_lib=[`find_lib libpng.dylib`]
                 ;;
-            *-*-cygwin* | *-*-mingw32*)
+            *-*-cygwin*)
+                png_lib="cygpng15-15.dll"
+                ;;
+            *-*-mingw32*)
                 png_lib=[`find_lib "libpng*.dll"`]
                 ;;
             *)
@@ -233,7 +240,10 @@ if test x$enable_tif = xyes; then
             *-*-darwin*)
                 tif_lib=[`find_lib libtiff.dylib`]
                 ;;
-            *-*-cygwin* | *-*-mingw32*)
+            *-*-cygwin*)
+                tif_lib="cygtiff-5.dll"
+                ;;
+            *-*-mingw32*)
                 tif_lib=[`find_lib "libtiff-*.dll"`]
                 ;;
             *)
